### PR TITLE
Update Message Media.

### PR DESCRIPTION
**Pull request by codizen**
- set filename using option or from url path.

### CODE REVIEW

File: src/structures/MessageMedia.js [hunk 1]
```diff
@@ -58,6 +59,13 @@ class MessageMedia {
      */
     static async fromUrl(url, options = {}) {
         let mimetype;
+        let filename = null
```
Comment (tuyuribr) [inline src/structures/MessageMedia.js:62]: Missing semicolon

File: src/structures/MessageMedia.js [hunk 2]
```diff
@@ -58,6 +59,13 @@ class MessageMedia {
...
+
+        if (!options.filename) {
+            filename = new URL(decodeURIComponent(url)).pathname.split('/').pop();
+        } else {
+            filename = options.filename
```
Comment (tuyuribr) [inline src/structures/MessageMedia.js:67]: Missing semicolon

File: src/structures/MessageMedia.js (hunk 1 above)
Comment (tuyuribr) [reply src/structures/MessageMedia.js:62]: Maybe doesnt need to be set to null

File: src/structures/MessageMedia.js [hunk 3]
```diff
@@ -58,6 +59,13 @@ class MessageMedia {
      */
     static async fromUrl(url, options = {}) {
         let mimetype;
+        let filename = null
+
+        if (!options.filename) {
```
Comment (tuyuribr) [inline src/structures/MessageMedia.js:64]: this can be simplified

File: src/structures/MessageMedia.js (hunk 1 above)
Comment (codizen) [reply src/structures/MessageMedia.js:62]: Ok, thanks to this check.

File: src/structures/MessageMedia.js (hunk 2 above)
Comment (codizen) [reply src/structures/MessageMedia.js:67]: Ok, thanks to this check.

**Comment by tuyuribr** [comment]
Thanks for your contribution, but I think #811 resolve this issue, right?

File: src/structures/MessageMedia.js (hunk 3 above)
Comment (codizen) [reply src/structures/MessageMedia.js:64]: Do you want to simplify in one line?

**Comment by codizen** [comment]
> Thanks for your contribution, but I think #811 resolve this issue, right?

Yes similar, but have a small difference.
In this code, we can pass filename in options and clean filename if get from url path.

**Comment by PurpShell** [comment]
> > Thanks for your contribution, but I think #811 resolve this issue, right?
> 
> Yes similar, but have a small difference.
> In this code, we can pass filename in options and clean filename if get from url path.

Hey! I appreciate you coming here and taking the effort to open a new PR. Please coordinate your efforts, this is not a marketplace. You can open a PR in a fork, so maybe port your changes there? As for setting a filename by an option, you can simply just change the property on the MessageMedia object.

```js
let media = await MessageMedia.fromURL("https://abcd.xyz");
media.filename = "test";
```

**Comment by PurpShell** [comment]
Update: Seems like #811 has implemented the options you were looking for. I'll close this PR, hopefully you continue contributing there!

**Pull request closed without merging**